src/zc/buildout/easy_install.py

```diff
@@ -1658,7 +1666,117 @@ def call_easy_install(spec, dest):
+        with open(ns_file) as f:
+            namespace_packages = [
+                line.strip().replace('.', os.path.sep)
+                for line in f.readlines()
+            ]
+
+        for namespace_package in namespace_packages:
+            init_py_file = os.path.join(dest, namespace_package, '__init__.py')
+            with open(init_py_file, 'w') as f:
+                f.write(
+                    "__import__('pkg_resources').declare_namespace(__name__)"
+                )
+
+    # Remove `bin` directory if needed
+    # as there is no way to avoid script installation
+    # when running `pip install`
+    ep_file = os.path.join(dest, distinfo_dir, 'entry_points.txt')
+    if os.path.exists(ep_file):
+        with open(ep_file) as f:
+            if "console_scripts" in f.read():
+                bin_dir = os.path.join(dest, 'bin')
+                if os.path.exists(bin_dir):
+                    shutil.rmtree(bin_dir)
+
+    # Make properly named new egg dir
+    distro = [d for d in pkg_resources.find_distributions(dest)][0]
+    egg_name = distro.egg_name() + '.egg'
+    egg_dir = os.path.join(dest, egg_name)
+    os.mkdir(egg_dir)
+
+    # Move ".dist-info" dir into new egg dir
+    os.rename(
+        os.path.join(dest, distinfo_dir),
+        os.path.join(egg_dir, distinfo_dir)
+    )
+
+    with open(os.path.join(egg_dir, distinfo_dir, 'top_level.txt')) as f:
+        top_levels = filter(
+            (lambda x: len(x) != 0),
+            [line.strip() for line in f.readlines()]
+            )
+
+    # Move all top_level modules or packages
+    for top_level in top_levels:
+        # as package
+        top_level_dir = os.path.join(dest, top_level)
+        if os.path.exists(top_level_dir):
+            shutil.move(top_level_dir, egg_dir)
+            continue
+        # as module
+        top_level_py = top_level_dir + '.py'
+        if os.path.exists(top_level_py):
+            shutil.move(top_level_py, egg_dir)
+            top_level_pyc = top_level_dir + '.pyc'
+            if os.path.exists(top_level_pyc):
+                shutil.move(top_level_pyc, egg_dir)
+            continue
+
+    def get_all_files(it):
+        r = csv.reader(FakeFile(it))
+        for row in r:
+            yield row[0]
+
+    with open(os.path.join(egg_dir, distinfo_dir, 'RECORD')) as f:
+        all_files = get_all_files(f.readlines())
+
+
+    # There might be some c extensions left over
+    for entry in all_files:
+        if _do_not_keep_leftover(entry):
+            continue
+        dest_entry = os.path.join(dest, entry)
+        egg_entry = os.path.join(egg_dir, entry)
+        if os.path.exists(dest_entry) and not os.path.exists(egg_entry):
+            os.rename(dest_entry, egg_entry)
+
+    return [egg_dir]
+
+
+SLASH_BIN_SLASH = os.path.join('', 'bin', '')
+
+
+def _do_not_keep_leftover(filename):
+    return (
+        filename.endswith('.pyc') or
+        filename.endswith('.pyo') or
+        SLASH_BIN_SLASH in filename
```

Comment:
Ooh this could bite us if there's ever a subpackage named `bin`.

(Also, what about Scripts on Windows?)